staging/src/k8s.io/apiextensions-apiserver/main.go

```diff
@@ -34,9 +36,15 @@ func main() {
 		runtime.GOMAXPROCS(runtime.NumCPU())
 	}
 
-	cmd := server.NewCommandStartCustomResourceDefinitionsServer(os.Stdout, os.Stderr, wait.NeverStop)
+	stopCh, exitCodeCh := genericapiserver.SetupSignalHandler()
+	exitCode := 1
+	defer func() { exitCodeCh <- exitCode }()
+
+	cmd := server.NewCommandStartCustomResourceDefinitionsServer(os.Stdout, os.Stderr, stopCh)
 	cmd.Flags().AddGoFlagSet(flag.CommandLine)
-	if err := cmd.Execute(); err != nil {
-		panic(err)
+	if err := cmd.Execute(); err == nil {
+		exitCode = 0
+	} else {
+		glog.Error(err.Error())
```

Comment:
That's nice, your `SetupSignalHandler` also nicely takes care of exiting with code != 0 :)